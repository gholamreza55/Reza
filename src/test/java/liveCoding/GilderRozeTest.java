package liveCoding;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

public class GilderRozeTest {
    @Test
    void should_update_quality() {
        String[] name = new String[]{"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert"};
        Integer[] selin = new Integer[]{0, -1, 1, 11, 12};
        Integer[] quality = new Integer[]{0, 1, 2, 50, 49, -1};
        //Item output = Updatability(foo, selin, quality);
        CombinationApprovals.verifyAllCombinations(this::Updatequlity, name, selin, quality);
    }

    private String Updatequlity(String name, int selin, int quality) {
        Item[] items = new Item[]{new Item(name, selin, quality)};
        GilderRoze app = new GilderRoze(items);
        app.updatequality();
       // Item output = app.items[0];
        return app.items[0].toString();
    }
}
