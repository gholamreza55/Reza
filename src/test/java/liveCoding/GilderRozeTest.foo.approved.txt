[foo, 0, 0] => Item{name='foo', selin=0, quality=0} 
[foo, 0, 1] => Item{name='foo', selin=0, quality=0} 
[foo, 0, 2] => Item{name='foo', selin=0, quality=1} 
[Aged Brie, 0, 0] => Item{name='Aged Brie', selin=0, quality=1} 
[Aged Brie, 0, 1] => Item{name='Aged Brie', selin=0, quality=2} 
[Aged Brie, 0, 2] => Item{name='Aged Brie', selin=0, quality=3} 
